src/panels/lovelace/cards/hui-markdown-card.ts

```diff
@@ -113,7 +114,18 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
     if (changedProps.has("_config")) {
       this._tryConnect();
     }
-
+    if (
+      this._templateResult &&
+      this._templateResult.result.length === 0 &&
+      this._config.show_empty === false
+    ) {
+      if (!this.hidden) {
+        this.style.display = "none";
+        this.toggleAttribute("hidden", true);
+        fireEvent(this, "card-visibility-changed", { value: false });
```

Comment:
You implemented logic to hide it, but there is no logic that unhides it ones the template does give a result?